apps/imgviewer.c

```diff
@@ -100,143 +116,174 @@ void resize_finish(int w, int h) {
 	yutani_flip(yctx, window);
 }
 
+int file_exists(char * filename){
+	return access(filename, F_OK);
+}
 
-int main(int argc, char * argv[]) {
+int is_readable(char * filename){	
 
-	static struct option long_opts[] = {
-		{"help",   no_argument,       0, '?'},
-		{0,0,0,0}
-	};
-
-	if (argc > 1) {
-		/* Read some arguments */
-		int index, c;
-		while ((c = getopt_long(argc, argv, "h", long_opts, &index)) != -1) {
-			if (!c) {
-				if (long_opts[index].flag == 0) {
-					c = long_opts[index].val;
-				}
-			}
-			switch (c) {
-				case 'h':
-					usage(argv);
-					exit(0);
-					break;
-				default:
-					break;
-			}
-		}
-	}
-
-	yctx = yutani_init();
-	if (!yctx) {
-		fprintf(stderr, "%s: failed to connect to compositor\n", argv[0]);
+	if ( file_exists( filename) != 0 ){
+		fprintf(stderr, "%s: not exist! error_info: %s\n", filename, strerror( errno ) );
 		return 1;
 	}
-	init_decorations();
+	
+	FILE * tmp = fopen( filename, "r" );
```

Comment:
`load_sprite` already calls `fopen` which is why I recommended a small change there instead of adding checks here.